### PR TITLE
fix cleanup of rebases in carstore

### PR DESCRIPTION
**Pull request by whyrusleeping**
This fixes two things, first the test-pds didnt properly implement rebases which caused some issues in tests. 
The second thing was that old blockRefs werent getting deleted after the rebase, despite their associated CarShards getting deleted.

### CODE REVIEW

File: indexer/crawler.go [hunk 1]
```diff
@@ -214,12 +215,19 @@ func (c *CrawlDispatcher) AddToCatchupQueue(ctx context.Context, host *models.PD
 		panic("not okay")
```
Comment (ericvolp12) [inline indexer/crawler.go:215]: Why do we panic here?

File: indexer/indexer.go [hunk 2]
```diff
@@ -838,28 +838,48 @@ func (ix *Indexer) FetchAndIndexRepo(ctx context.Context, job *crawlWork) error
 	}
 
 	var rebase *comatproto.SyncSubscribeRepos_Commit
-	for _, job := range job.catchup {
-		if job.evt.Rebase {
-			rebase = job.evt
+	var rebaseIx int
+	for i, j := range job.catchup {
+		if j.evt.Rebase {
+			fmt.Println("REBASE AT: ", i, len(job.catchup))
```
Comment (ericvolp12) [inline indexer/indexer.go:844]: Do we have a logger we can use for this?

**Review by ericvolp12 — APPROVED** [review]
This looks okay to me but I've also not got a ton of context on the indexer code yet.

File: indexer/crawler.go (hunk 1 above)
Comment (whyrusleeping) [reply indexer/crawler.go:215]: I should make those messages better... but we need to be aware of which PDS the work is coming from for scheduling purposes

File: indexer/indexer.go (hunk 2 above)
Comment (whyrusleeping) [reply indexer/indexer.go:844]: oh whoops, yes.

**Pull request merged**